app/src/main/java/com/example/myapplication/MainActivity.java

```diff
@@ -1,17 +1,55 @@
 package com.example.myapplication;
 
-import android.content.Context;
+import android.app.Activity;
 import android.content.Intent;
 import android.os.Bundle;
+import android.util.Log;
 import android.widget.Button;
+import android.widget.TextView;
 
+import androidx.activity.result.ActivityResult;
+import androidx.activity.result.ActivityResultCallback;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
 import androidx.appcompat.app.AppCompatActivity;
 
-import com.example.testlibrary.CallbackService;
-import com.example.testlibrary.LibraryService;
+import com.example.testlibrary.LibraryActivity;
 
 public class MainActivity extends AppCompatActivity {
 
+    private static final String TAG = "MainActivity";
+
+    ActivityResultLauncher<Intent> activityLauncher = registerForActivityResult(
+            new ActivityResultContracts.StartActivityForResult(),
+            new ActivityResultCallback<ActivityResult>() {
+                @Override
+                public void onActivityResult(ActivityResult result) {
+                    TextView resultTextView = findViewById(R.id.resultTextView);
+                    int resultCode = result.getResultCode();
+
+                    String message = null;
+                    switch (resultCode) {
+                        case Activity.RESULT_OK:
+                            Intent intent = result.getData();
+                            if (intent != null) {
+                                message = intent.getStringExtra(LibraryActivity.EXTRA_MESSAGE);
+                            }
+                            break;
+                        case Activity.RESULT_CANCELED:
+                            message = "CANCELED";
+                            break;
+                        default:
+                            message = "UNKNOWN";
+                    }
+
+                    String resultText = String.format("Result: %s", message) ;
+                    Log.d(TAG, resultText);
+
+                    // also display result on screen
+                    resultTextView.setText(resultText);
+                }
+            });
```

Comment:
The `onActivityResultLogic` displayed here can be further simplified by the SDK with a custom [ActivityResultContract](https://developer.android.com/training/basics/intents/result#custom).